libp2p/switch.nim

```diff
@@ -375,17 +375,24 @@ proc stop*(s: Switch) {.async.} =
     warn "error stopping switch", exc = exc.msg
 
 proc subscribeToPeer*(s: Switch, peerInfo: PeerInfo) {.async, gcsafe.} =
+  trace "about to subscribe to pubsub peer", peer = peerInfo.shortLog()
   ## Subscribe to pub sub peer
-  if s.pubSub.isSome and peerInfo.id notin s.dialedPubSubPeers:
+  if s.pubSub.isSome and (peerInfo.id notin s.dialedPubSubPeers):
+    let conn = await s.getMuxedStream(peerInfo)
     try:
-      s.dialedPubSubPeers.incl(peerInfo.id)
-      let conn = await s.dial(peerInfo, s.pubSub.get().codec)
       if isNil(conn):
```

Comment:
this goes outside the try because logically, conn must not be null in the try block, in general